nautobot/extras/models/relationships.py

```diff
@@ -674,3 +676,32 @@ def clean(self):
                             )
                         }
                     )
+
+        if self.relationship.destination_filter or self.relationship.source_filter:
+            self._validate_relationship_filter_restriction()
+
+    def _validate_relationship_filter_restriction(self):
+        """Validate relationship association do not violate filter restrictions"""
+        sides = []
+
+        if self.relationship.destination_filter:
+            sides.append("destination")
+
+        if self.relationship.source_filter:
+            sides.append("source")
+
+        for side_name in sides:
+            side = getattr(self, side_name)  # destination / source
+            side_filter = getattr(self.relationship, f"{side_name}_filter")
+            try:
+                filterset_class = get_filterset_class_for_model(side.__class__)
```

Comment:
Here is where we could instead just have a check where `if filterset_class is None` using the existing function, instead of catching the exception from the new function?